.github/workflows/on-push-to-main.yml

```diff
@@ -18,6 +18,10 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: cachix/install-nix-action@v24
+        with:
+          extra_nix_config: |
+            sandbox = false
```

Comment:
Is this about https://discourse.nixos.org/t/what-is-sandboxing-and-what-does-it-entail/15533?

Reply:
Yes, we need internet access.

Reply:
Thank you.

If I understand correctly:
- With this PR, we will pull in the latest available `nightly` version of Rust, and use it for building `query-engine-wasm`
- In the future, we will try to use a fixed `nightly` version of Rust, whose version we will bump from time to time (like we do for our target `stable` version of Rust)

Is that right?

Reply:
(Uhm, but I see on https://github.com/prisma/prisma-engines/pull/4682 that we actually use `rustup default nightly-2024-01-25` in the `build.sh` file. I'm confused about when we use Nix vs `rustup`, and whether Nix influences or uses `rustup` commands in any way)

Reply:
@jkomyno ideally later we'll use nix-installed nightly on CI so we can update it together with stable in a single `nix flake update` but we needed to make it work with rustup as well to make the local script usable for those who don't use nix, so for now it always uses rustup to make things simple and avoid the conditional logic in the build script, so this is shippable already without spending too much time.

Reply:
Thank you for the clarification!

Reply:
Also we can't have pure Nix build as a Nix derivation anyway because `-Zbuild-std` is currently incompatible with `cargo vendor` (unless you depend on `compiler_builtins` explicitly as a cargo crate but then things become messy). So even with a Nix-installed toolchain we would still have to do an impure build without sandboxing until the issue is fixed upstream in cargo.